src/software/geom/shot.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "software/geom/point.h"
+
+class Shot
+{
+   public:
+    /**
+     * Creates a shot with the given target and angle
+     *
+     * @param point The target to shoot at
+     * @param angle The angle between the obstacles on either side of the shot vector
+     *
+     */
+    Shot(Point point, Angle angle);
+
+    /**
+     * Returns a point, the target to shoot at.
+     * @return a point, representing the target to shoot at.
+     */
+    Point getPoint() const;
+
+    /**
+     * Returns an angle between the obstacles on either side of the shot vector.
+     * @return an angle between the obstacles on either side of the shot vector.
+     */
```

Comment:
I might phrase it as: `the angle formed by the shot origin, and the two obstacles closest to the shot path`, but this is a tricky thing to phrase so I'm not totally sure on that.......... anyone else have a strong opinion? @MathewMacDougall ?

Reply:
Maybe `the angle formed by the shot origin, and the edged of the two obstacles closest to the shot path`? Or when in doubt, add some ASCII art!

Reply:
Checkout www.asciiflow.com for making nice ascii art

Reply:
I added the new Angle definition and a diagram to the Angle.h file. Not sure if my interpretation of Angle is correct or if this diagram is any good but here it is. Let me know!

Reply:
Looks great!